raft/node.go

```diff
@@ -224,10 +224,15 @@ func RestartNode(c *Config) Node {
 	return &n
 }
 
+type proposingMsg struct {
```

Comment:
the message type inside this will not always be proposal.

renaming this to nodeMsg or msgWithResult is better.